libtcmu_log.c

```diff
@@ -554,6 +582,14 @@ static void *log_thread_start(void *arg)
 	}
 	pthread_mutex_unlock(&tcmu_logbuf->lock);
 
+	pthread_mutex_lock(&log_tmpbuf_lock);
+	darray_foreach(lb, log_tmpbuf)
+		log_output(lb->pri, lb->msg, true);
+	darray_free(log_tmpbuf);
```

Comment:
I think we have a couple leaks.

1. If we never get the thread started we never free the log_tmpbuf.

2. In main.c's main failure handling path we do:

```
destroy_log:
        tcmu_destroy_log();
destroy_config:
        tcmu_destroy_config(tcmu_cfg);

```

So the goto path in main.c that calls tcmu_destroy_config path could do 
tcmu_destroy_config -> tcmu_cancel_config_thread -> ... log_internal

Reply:
I think we can fix this by doing:

1. Just clear log_tmpbuf under the lock after we free it above. Then in log_cleanup add a check for if it is still set. If so then free it.

2. If log_tmpbuf is cleared then just drop down to vfprintf(stderr, fmt, args); calls in log_internal.